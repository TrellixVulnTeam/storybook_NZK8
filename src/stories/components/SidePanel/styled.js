// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import { lighten } from 'polished';

const SIDE_MENU_WIDTH = 260;
const OVERLAY_OFFSET = 50;
/**
 * Determine if offset for transform should
 * be positive on negative based on side prop
 * @param {string} side
 * @param {string} width
 * @return {number}
 */
function generateOffset(side, width) {
  if (width === '100%') {
    // eslint-disable-next-line no-restricted-globals
    return side === 'left' ? `-${screen.width}` : screen.width;
  }

  return side === 'left' ? `-${width}` : width;
}

/**
 * Determine if box-shadow should
 * be offset from the left or right
 * based on side prop
 * @param {string} side
 * @return {string}
 */
function generateBoxShadow(side) {
  return side === 'left' ? '2px 7px 10px rgba(0,0,0,0.2)' : '-5px 0 30px rgba(0, 0, 0, 0.2)';
}

/* eslint-disable no-restricted-globals */
export const Wrapper = styled.div`
  position: fixed;
  overflow: hidden;
  top: 0;
  right: ${p => p.side === 'right' && 0};
  left: ${p => p.side === 'left' && 0};
  width: ${p =>
    p.width === '100%' ? `calc(100% - ${SIDE_MENU_WIDTH + OVERLAY_OFFSET}px)` : `${p.width}px`};
  height: 100%;
  min-height: 500px;
  display: flex;
  flex-direction: column;
  background: ${colors.white};
  box-shadow: ${p => p.show && generateBoxShadow(p.side)};
  z-index: 11;
  transition: all 300ms ease;
  transform: translateX(${p => (p.show ? 0 : generateOffset(p.side, p.width))}px);
  transition-property: width, transform;

  @media (min-width: 1200px) {
    left: ${p => p.side === 'left' && '260px'};
  }

  @media (max-width: 1200px) {
    width: ${p => (p.width === '100%' ? `calc(100% - ${OVERLAY_OFFSET}px)` : `${p.width}px`)};
  }

  @media (max-width: 500px) {
    width: ${screen.width}px;
    transform: translateX(${p => (p.show ? 0 : generateOffset(p.side, '100%'))}px);
  }
`;
/* eslint-enable no-restricted-globals */

Wrapper.defaultProps = {
  side: 'right',
  width: 500,
};

export const Header = styled.div`
  display: inline-flex;
  justify-content: space-between;
  margin-top: 0;
  padding: 20px;
  width: 100%;
  height: 60px;
  background: #eee;
  font-size: 16px;
  font-weight: 600;
  color: ${colors.steel};
`;

export const InnerWrapper = styled.div`
  flex: 1;
  width: '100%';
  overflow-y: auto;
  overflow-x: hidden;

  /* scrollbar */
  ::-webkit-scrollbar {
    width: 7px;
  }
  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: #4d5259;
  }
  &:hover {
    ::-webkit-scrollbar-thumb {
      background-color: ${lighten(0.2, '#4d5259')};
    }
  }
`;
