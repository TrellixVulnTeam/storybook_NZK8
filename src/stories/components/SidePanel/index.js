// @flow

import React from 'react';
import SidePanelHeader from 'components/SidePanelHeader';
import { Wrapper, InnerWrapper } from './styled';

type Props = {
  children: React$Node,
  show: boolean,
  side?: 'left' | 'right',
  width?: string,
  header?: React$Node | boolean,
  headerNav?: { title: string, handleOnClick: Function },
  headerTitle?: string,
  headerButtons?: React$Node,
  // eslint-disable-next-line react/no-unused-prop-types
  onClose?: Function,
};

export default (props: Props) => (
  <Wrapper width={props.width} show={props.show} side={props.side}>
    {(() => {
      if (props.header === false) {
        return null;
      }
      if (props.header === undefined) {
        if (!props.headerNav && !props.headerTitle && !props.headerButtons) {
          return null;
        }

        return (
          <SidePanelHeader
            nav={props.headerNav}
            title={props.headerTitle || ''}
            buttons={props.headerButtons}
          />
        );
      }
      return props.header;
    })()}
    <InnerWrapper>{props.children}</InnerWrapper>
  </Wrapper>
);
