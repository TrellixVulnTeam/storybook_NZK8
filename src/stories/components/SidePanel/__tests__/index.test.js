import React from 'react';
import { mount } from 'enzyme';
import SidePanel from 'components/SidePanel/index';

describe('<SidePanel />', () => {
  const mandatoryProps = {
    show: true,
  };

  it('should render correctly with mandatory props', () => {
    const sut = mount(<SidePanel {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if not show and full props', () => {
    const props = {
      ...mandatoryProps,
      show: false,
      side: 'left',
      header: false,
    };
    const sut = mount(<SidePanel {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with header', () => {
    const props = {
      ...mandatoryProps,
      header: true,
      side: 'left',
    };
    const sut = mount(<SidePanel {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with header title, right side and 100% width when no show', () => {
    const props = {
      ...mandatoryProps,
      show: false,
      width: '100%',
      headerTitle: 'title',
      side: 'right',
    };
    const sut = mount(<SidePanel {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with headerButtons', () => {
    const props = {
      ...mandatoryProps,
      headerButtons: [() => {}],
    };
    const sut = mount(<SidePanel {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with right side if no show', () => {
    const props = {
      ...mandatoryProps,
      show: false,
      side: 'right',
    };
    const sut = mount(<SidePanel {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
