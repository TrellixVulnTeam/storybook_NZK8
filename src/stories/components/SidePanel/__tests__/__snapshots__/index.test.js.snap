// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidePanel /> should render correctly if not show and full props 1`] = `
<_default
  header={false}
  show={false}
  side="left"
>
  <styled__Wrapper
    show={false}
    side="left"
    width={500}
  >
    <div
      className="styled__Wrapper-pphozb-0 fjHRVi"
      width={500}
    >
      <styled__InnerWrapper>
        <div
          className="styled__InnerWrapper-pphozb-2 hsrEzi"
        />
      </styled__InnerWrapper>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<SidePanel /> should render correctly with header 1`] = `
<_default
  header={true}
  show={true}
  side="left"
>
  <styled__Wrapper
    show={true}
    side="left"
    width={500}
  >
    <div
      className="styled__Wrapper-pphozb-0 boaNwF"
      width={500}
    >
      <styled__InnerWrapper>
        <div
          className="styled__InnerWrapper-pphozb-2 hsrEzi"
        />
      </styled__InnerWrapper>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<SidePanel /> should render correctly with header title, right side and 100% width when no show 1`] = `
<_default
  headerTitle="title"
  show={false}
  side="right"
  width="100%"
>
  <styled__Wrapper
    show={false}
    side="right"
    width="100%"
  >
    <div
      className="styled__Wrapper-pphozb-0 ecXJV"
      width="100%"
    >
      <_default
        title="title"
      >
        <styled__Wrapper>
          <div
            className="styled__Wrapper-sc-10wtz8u-0 eoaLjo"
          >
            <styled__NavWrapper>
              <div
                className="styled__NavWrapper-sc-10wtz8u-1 eanWvd"
              />
            </styled__NavWrapper>
            <styled__TitleWrapper>
              <div
                className="styled__TitleWrapper-sc-10wtz8u-2 jhnXZS"
              >
                title
              </div>
            </styled__TitleWrapper>
            <styled__ButtonsWrapper>
              <div
                className="styled__ButtonsWrapper-sc-10wtz8u-3 fWxmrY"
              />
            </styled__ButtonsWrapper>
          </div>
        </styled__Wrapper>
      </_default>
      <styled__InnerWrapper>
        <div
          className="styled__InnerWrapper-pphozb-2 hsrEzi"
        />
      </styled__InnerWrapper>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<SidePanel /> should render correctly with headerButtons 1`] = `
<_default
  headerButtons={
    Array [
      [Function],
    ]
  }
  show={true}
>
  <styled__Wrapper
    show={true}
    side="right"
    width={500}
  >
    <div
      className="styled__Wrapper-pphozb-0 bJLdMT"
      width={500}
    >
      <_default
        buttons={
          Array [
            [Function],
          ]
        }
        title=""
      >
        <styled__Wrapper>
          <div
            className="styled__Wrapper-sc-10wtz8u-0 eoaLjo"
          >
            <styled__NavWrapper>
              <div
                className="styled__NavWrapper-sc-10wtz8u-1 eanWvd"
              />
            </styled__NavWrapper>
            <styled__TitleWrapper>
              <div
                className="styled__TitleWrapper-sc-10wtz8u-2 jhnXZS"
              />
            </styled__TitleWrapper>
            <styled__ButtonsWrapper>
              <div
                className="styled__ButtonsWrapper-sc-10wtz8u-3 fWxmrY"
              />
            </styled__ButtonsWrapper>
          </div>
        </styled__Wrapper>
      </_default>
      <styled__InnerWrapper>
        <div
          className="styled__InnerWrapper-pphozb-2 hsrEzi"
        />
      </styled__InnerWrapper>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<SidePanel /> should render correctly with mandatory props 1`] = `
<_default
  show={true}
>
  <styled__Wrapper
    show={true}
    side="right"
    width={500}
  >
    <div
      className="styled__Wrapper-pphozb-0 bJLdMT"
      width={500}
    >
      <styled__InnerWrapper>
        <div
          className="styled__InnerWrapper-pphozb-2 hsrEzi"
        />
      </styled__InnerWrapper>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<SidePanel /> should render correctly with right side if no show 1`] = `
<_default
  show={false}
  side="right"
>
  <styled__Wrapper
    show={false}
    side="right"
    width={500}
  >
    <div
      className="styled__Wrapper-pphozb-0 gHArpk"
      width={500}
    >
      <styled__InnerWrapper>
        <div
          className="styled__InnerWrapper-pphozb-2 hsrEzi"
        />
      </styled__InnerWrapper>
    </div>
  </styled__Wrapper>
</_default>
`;
